manifests/init.pp

```diff
@@ -64,14 +67,16 @@
   $package_name            = $nsclient::params::package_name,
   $download_destination    = $nsclient::params::download_destination,
   $config_template         = $nsclient::params::config_template,
-  $install_path            = $nsclient::params::install_path
+  $install_path            = $nsclient::params::install_path,
+  $password                = $nsclient::params::password
 ) inherits nsclient::params {
 
   validate_string($package_source_location)
   validate_string($package_source)
   validate_string($package_name)
   validate_string($config_template)
   validate_string($install_path)
+  validate_string($password)
```

Comment:
Please do not use the legacy validate_*() functions. Instead we can rely on datatypes.